_posts/2019-04-03-blog-import-the-style-to-the-typescript-file.md

```diff
@@ -0,0 +1,112 @@
+
+```javascript
+{
+  loader: 'typings-for-css-modules-loader',
+    options: {
+      modules: true,
+      localIdentName: `${hashPrefix}_[path]_[name]_[local]_[hash:base64:5]`,
+      namedExport: true,
+      camelCase: true,
+			banner: '// *** Generated File - Do not Edit ***'
+    }
+}
+```
+However, it has some flaw. The process of the definition file generator need always start a watch by a command like `yarn start`. It also makes our size of the repository become a bigger one although the intellisense is convenient for us.
+
+
+
+### Using the "require" syntax instead of using import.
+
+Using the different style or syntax for import may make our project become a mess, we should keep a consistent form of the import, so maybe there were have a better solution that I not found yet.
+
+
+
+### TypeScript server plugin - [typescript -plugin-css-modules](https://github.com/mrmckeb/typescript-plugin-css-modules)
+This plugin is a TypeScript plugin was inspired by this [create-react-app issue](https://github.com/facebook/create-react-app/issues/5677) and based on css-module-types. It's easy to set up by installing the node module and add plugin to the `tsconfig.json` like the following:
+
+```json
+{
+  "compilerOptions": {
+    "plugins": [{ "name": "typescript-plugin-css-modules" }]
+  }
+}
+```
+
+The match pattern default is `\\.module\\.(sa|sc|c)ss$`. We can custom set other pattern by the property.
+
+```json
+{
+  "compilerOptions": {
+    "plugins": [
+      {
+        "name": "typescript-plugin-css-modules",
+        "options": {
+          "customMatcher": "\\.m\\.css$",
+          "camelCase": "dashes"
+        }
+      }
+    ]
+  }
+}
+```
+After the setting, we also have to change the version of TypeScript. By default, VSCode will use its own version of TypeScript. To make it work correctly with this plugin, there have two options:
+
+1. Add this plugin to "typescript.tsserver.pluginPaths" in settings.
+```json
+{
+  "typescript.tsserver.pluginPaths": ["typescript-plugin-css-modules"]
+}
+```
+But this option is only setable for users setting. It's not convinient for coporation. The next option is more useful for us.
+
+2. Use our workspace's version of TypeScript
+  The workspace's version which will load the plugins from our tsconfig.json` file. We can add the config to the workspace setting.
+
+```json
+{
+  "typescript.tsdk": "./node_modules/typescript/lib"
+}
+```
+
+Besides, we found the issue about import not only style file but also other files like SVG in TypeScript. It will show the error like: "Cannot find module '../../assets/images/logo.svg'.ts(2307)". To resolve the issue that unable to import the SVG file in TypeScript, we can create a definition file named "custom.d.ts" and add the following code and add the file path to the includes field of the `tsconfig.json`:
```

Comment:
Aside from style files, we also run into the same issue for other files like SVG. It will show errors like: "Cannot find module '../../assets/images/logo.svg'.ts(2307)". To resolve this issue with loading SVG files, we can create a definition file named "custom.d.ts" with the code shown below, and add the file path to the "includes" field in the `tsconfig.json` file.